File: Fall-2016-17/week3_disemvoweler/disemvowel.py
#!/usr/bin/env python3
"""
Disemvoweler: Python 3 Solution
Author: Peter Gagliardi
"""

def disemvowel(sentence):
    vowels = ""
    consonants = ""
    for c in sentence.lower():
        if c in "aeiou":
            vowels += c
        elif c == ' ':
            continue
        else:
            consonants += c
    return vowels, consonants

if __name__ == "__main__":
    words = input('Enter some words: ')
    vowels, consonants = disemvowel(words)
    print("After the operation: {}".format(consonants))
    print("Removed vowels: {}".format(vowels))

File: Fall-2016-17/week3_disemvoweler/disemvowel.rb
=begin
Disemvoweler Ruby Solution
Author: Peter Gagliardi
=end

#
# Take a string and divide it into two strings: one that contains only
# the vowels and another for consonants
#
def disemvowel sentence
    # Convert to lowercase and strip out spaces
    input = sentence.downcase.tr ' ', ''

    # Split the characters into [[vowels], [consonants]] and then convert
    # to two strings
    input.chars
        .partition {|x| "aeiou ".include? x}
        .map {|x| x.join ''}
end

# Test it out
puts "Time to remove some vowels:"
puts "Enter some words separated by spaces"
words = gets.chomp
vowels, consonants = disemvowel words
puts "Now without vowels: #{consonants}\nWe removed: #{vowels}"
